segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -240,10 +248,21 @@ SnapshotInfoStore getStorageSnapshotInfoStore() {
                 () -> readStorageSnapshot(config.getStorageSnapshotTimeout()));
     }
 
+    private Function<String, CompletableFuture<DirectSegmentAccess>> getOrCreateContainerProcessorSegment(int containerId,
+                                                                                                          Duration duration) {
+        final SegmentType systemCritical = SegmentType.builder().system().critical().build();
+        return s -> Futures.exceptionallyComposeExpecting(
+                this.forSegment(NameUtils.getEventProcessorSegmentName(containerId, s), duration),
+                e -> e instanceof StreamSegmentNotExistsException,
+                () -> createStreamSegment(NameUtils.getEventProcessorSegmentName(containerId, s), systemCritical, null, duration)
```

Comment:
Just to confirm - this will create a segment for each container + consumer, right? (nothing to do here - just a question)

Reply:
Correct.